cli/Valet/Diagnose.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+namespace Valet;
+
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Output\ConsoleOutput;
+
+class Diagnose
+{
+    var $commands = [
+        'sw_vers',
+        'valet --version',
+        'cat ~/.config/valet/config.json',
+        'cat ~/.composer/composer.json',
+        'composer global diagnose',
+        'ls -al /etc/sudoers.d/',
+        'brew config',
+        'brew services list',
+        'brew list --versions | grep -E "(php|nginx|dnsmasq|mariadb|mysql|mailhog|openssl)(@\d\..*)?\s"',
+        'php -v',
+        'which -a php',
+        'php --ini',
+        'nginx -v',
+        'curl --version',
+        'php --ri curl',
+        '~/.composer/vendor/laravel/valet/bin/ngrok version',
+        'ls -al ~/.ngrok2',
+        'brew info nginx',
+        'brew info php',
+        'brew info openssl',
+        'openssl version -a',
+        'openssl ciphers',
+        'sudo nginx -t',
+        'which -a php-fpm',
+        'sudo /usr/local/opt/php/sbin/php-fpm --test',
```

Comment:
Do these `sudo` commands work when you're running this using `runAsUser`?  If I remember correctly, `runAsUser` is piping the command into a `sudo` command already, so I just want to make sure that's not getting mixed up or anything.

Reply:
I tried this initially but ended up reverting to prefixing with `sudo` because I wasn't getting the same results without it (reverted here https://github.com/laravel/valet/pull/938/commits/5baccdbb5427196d66857e3101beba214dd7b0fa). I would very much prefer remove these `sudo`s to stop consequent password prompts mid-run, will try give this another look with fresh eyes and report back if I find a better solution.

Reply:
Ok, did some more digging... Tested the `nginx -t` command: 

```bash
# without sudo, it doesn't work -- permission issues
nginx -t 

# this is what runAsUser() is calling and doesn't work -- permission issues
sudo -u [my username] nginx -t 

# works
sudo nginx -t 
```

The error for both commands that don't work is:

```
nginx: [alert] could not open error log file: open() "/usr/local/var/log/nginx/error.log" failed (13: Permission denied)
2020/04/26 13:13:30 [warn] 15096#0: the "user" directive makes sense only if the master process runs with super-user privileges, ignored in /usr/local/etc/nginx/nginx.conf:1
nginx: the configuration file /usr/local/etc/nginx/nginx.conf syntax is ok
2020/04/26 13:13:30 [emerg] 15096#0: open() "/usr/local/var/run/nginx.pid" failed (13: Permission denied)
nginx: configuration file /usr/local/etc/nginx/nginx.conf test failed
```

I imagine that running most commands using `runAsUser` is preferred to ensure the correct output, so will push a minor tweak to check for `sudo` in the command and use the basic `run` command in those cases.

Let me know if you have any other suggestions!

Reply:
This should now be fixed in https://github.com/laravel/valet/pull/938/commits/dedaddeff0bacda89908c1b83563ce133c94ced9